File: evals/refs/dot.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

func main() {
	in := bufio.NewScanner(os.Stdin)
	in.Buffer(make([]byte, 1<<20), 1<<20)
	in.Split(bufio.ScanWords)
	out := bufio.NewWriter(os.Stdout)
	defer out.Flush()

	read := func() string { in.Scan(); return in.Text() }

	T, _ := strconv.Atoi(read())
	for c := 0; c < T; c++ {
		N, _ := strconv.Atoi(read())
		a := make([]int64, N)
		b := make([]int64, N)
		for i := 0; i < N; i++ {
			v, _ := strconv.ParseInt(read(), 10, 64)
			a[i] = v
		}
		for i := 0; i < N; i++ {
			v, _ := strconv.ParseInt(read(), 10, 64)
			b[i] = v
		}
		var s int64 = 0
		for i := 0; i < N; i++ {
			s += a[i] * b[i]
		}
		fmt.Fprintln(out, s)
	}
}

File: evals/refs/dot.py
import sys
data = sys.stdin.read().split()
i = 0
T = int(data[i]); i += 1
out = []
for _ in range(T):
    N = int(data[i]); i += 1
    a = [int(data[i+j]) for j in range(N)]; i += N
    b = [int(data[i+j]) for j in range(N)]; i += N
    s = 0
    for k in range(N):
        s += a[k] * b[k]
    out.append(str(s))
sys.stdout.write('\n'.join(out) + ('\n' if out else ''))
